### PR TITLE
py sol for Planets and Kingdoms

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC: 
Screenshot from 2024-11-14 13-14-24
Screenshot from 2024-11-14 13-14-31

### CODE REVIEW

File: content/6_Advanced/SCC.mdx [hunk 1]
```diff
@@ -183,6 +183,86 @@ public class Main {
...
+			id[node] = comp
+			for u in adj_t[node]:
+				if not vis[u]:
+					stack.append(u)
+
+
+n, m = map(int, input().split())
+
+adj = [[] for _ in range(n)]
+adj_t = [[] for _ in range(n)]  # adj_t is the transpose of adj
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:237]: is this like the reversed adjacency list

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/SCC.mdx (hunk 1 above)
Comment (freakin23) [reply content/6_Advanced/SCC.mdx:237]: yes

File: content/6_Advanced/SCC.mdx [hunk 2]
```diff
@@ -183,6 +183,86 @@ public class Main {
...
+comps = 0
+
+for i in range(len(order) - 1, -1, -1):
+	v = order[i]
+	if not vis[v]:
+		comps += 1  # counter for SCC
+		dfs2(v, comps)
+
+print(comps)
+print(*id)
```
[the remark this answers is not in the document]
Comment (freakin23) [reply content/6_Advanced/SCC.mdx]: done

File: content/6_Advanced/SCC.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/6_Advanced/SCC.mdx:237]: rename it to `rev_adj`, much more intuitive than "transpose"

Comment (freakin23) [reply content/6_Advanced/SCC.mdx]: Done

**Comment by SansPapyrus683** [comment]
show proof of ac

**Comment by freakin23** [comment]
> show proof of ac

Screenshot_2024-11-14-23-05-40-45_3aea4af51f236e4932235fdada7d1643
Screenshot_2024-11-14-23-05-51-88_3aea4af51f236e4932235fdada7d1643
Screenshot_2024-11-14-23-05-57-71_3aea4af51f236e4932235fdada7d1643

**Pull request merged**